analyzers/src/SonarAnalyzer.CSharp/Rules/LoopsAndLinq.cs

```diff
@@ -0,0 +1,137 @@
+                _ => null
+            };
+
+        private static bool CanIfStatementBeMoved(IfStatementSyntax ifStatementSyntax) =>
+            ifStatementSyntax.Else == null
+            && ifStatementSyntax.Condition is InvocationExpressionSyntax invocationExpressionSyntax
+            && !invocationExpressionSyntax.DescendantNodes()
+                                          .OfType<ArgumentSyntax>()
+                                          .Any(argument => argument.RefOrOutKeyword.IsAnyKind(SyntaxKind.OutKeyword, SyntaxKind.RefKeyword));
+
+        private static void CheckIfCanBeSimplifiedUsingSelect(ForEachStatementSyntax forEachStatementSyntax, SyntaxNodeAnalysisContext c)
+        {
+            // There are multiple scenarios where the code can be simplified using LINQ.
+            // For simplicity, in the first version of the rule, we consider that Select() can be used
+            // only when a single property from the foreach variable is used.
+            // The issue is raised if:
+            //  - the property is used more than once
+            //  - the property is the right side of a variable declaration
+            var declaredSymbol = c.SemanticModel.GetDeclaredSymbol(forEachStatementSyntax);
+            var accessedProperties = new Dictionary<ISymbol, UsageStats>();
+
+            foreach (var identifierSyntax in GetIdentifiers(forEachStatementSyntax))
+            {
+                if (identifierSyntax.Parent is MemberAccessExpressionSyntax { Parent: not InvocationExpressionSyntax } memberAccessExpressionSyntax
+                    && !(identifierSyntax.Parent.Parent is AssignmentExpressionSyntax assignment && assignment.Left == identifierSyntax.Parent)
+                    && c.SemanticModel.GetSymbolInfo(identifierSyntax).Symbol.Equals(declaredSymbol))
+                {
+                    var symbol = c.SemanticModel.GetSymbolInfo(memberAccessExpressionSyntax.Name).Symbol;
+                    var usageStats = accessedProperties.GetOrAdd(symbol, _ => new UsageStats());
+
+                    if (identifierSyntax.Parent.Parent is EqualsValueClauseSyntax { Parent: VariableDeclaratorSyntax })
+                    {
+                        usageStats.IsInVarDeclarator = true;
+                    }
+
+                    usageStats.Count++;
+                }
+                else
+                {
+                    return;
+                }
+            }
+
+            if (accessedProperties.Count == 1
+                && accessedProperties.First().Value is var stats
+                && (stats.IsInVarDeclarator || stats.Count > 1))
+            {
+                var diagnostic = Diagnostic.Create(Rule,
+                                                   forEachStatementSyntax.Expression.GetLocation(),
+                                                   string.Format(SelectMessageFormat, forEachStatementSyntax.Identifier.ValueText, accessedProperties.Single().Key.Name));
```

Comment:
you don't need string.Format here, you just pass the message followed by the arguments and the `Create` method does the formatting, right?

Reply:
In this case, the main message is "{0}", which gets replaced by the generic one or the specific "select". Then, the select message is "Loop should be simplified by calling Select({0} => {0}.{1}))" for which we need the string format to apply the arguments.